types/src/test/java/org/infinispan/protostream/types/java/MarshallingTest.java

```diff
@@ -0,0 +1,231 @@
+package org.infinispan.protostream.types.java;
+
+import org.infinispan.protostream.GeneratedSchema;
+import org.infinispan.protostream.ImmutableSerializationContext;
+import org.infinispan.protostream.ProtobufUtil;
+import org.infinispan.protostream.SerializationContext;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class MarshallingTest {
```

Comment:
since there is another class called MarshallingTest, I would change the name of the class for something else